view/frontend/layout/rp_auction_index_index.xml

```diff
@@ -1,6 +1,10 @@
 <?xml version="1.0"?>
 <page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" layout="1column" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
     <referenceContainer name="content">
-        <block class="Magento\Framework\View\Element\Template" name="rp_auction_index_index" template="SomethingDigital_AuctionItem::index.phtml" />
+        <block class="Magento\Framework\View\Element\Template" name="rp_auction_index_index" template="SomethingDigital_AuctionItem::index.phtml">
+            <arguments>
+                <argument name="view-model" xsi:type="object">SomethingDigital\AuctionItem\ViewModel\AuctionItemCollectionRenderer</argument>
```

Comment:
nit: Convention with XML files in Magento is to use `snake_case`. This way it can properly generate the magic getters for the block class and you can do `$block->getViewModel()` in the template.